index.js

```diff
@@ -109,7 +120,12 @@ TChannel.prototype.listen = function () {
 		throw new Error('Missing server port.');
 	}
 
-	this.serverSocket.listen(this.port, this.host);
+	if (options.fd >= 0) {
+		this.listenFd = options.fd;
```

Comment:
agreed. that it is good to leave. just wanted to make sure that was the intention.